subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/ToBeFixedForInstantExecutionExtension.groovy

```diff
@@ -17,207 +17,181 @@
-    }
-
-    static boolean isAllIterations(String[] iterationMatchers) {
-        return iterationMatchers.length == 0
-    }
-
     @Override
     void visitFeatureAnnotation(ToBeFixedForInstantExecution annotation, FeatureInfo feature) {
-        // This override is required to satisfy spock's zealous runtime checks
-    }
-
-    /**
-     * Spock spec listener that registers failure collecting method interceptors for the given feature.
-     *
-     * Registration happens as late as possible and to the utmost position in the stack in order to catch
-     * failures without being sensible to other spock extensions that re-order interceptors nor to the place
-     * of annotated spock features in hierarchies of spec classes.
-     *
-     * This is also so that failures in setup and cleanup methods are taken into account by the annotation.
-     * Unfortunately this leads to not being able to throw from inside the interceptors but instead throw
-     * from outside the test execution, after setup/test/cleanup, in Spock's core, effectively stopping the
-     * whole spec execution. In practice, as soon as a test doesn't behave, we stop executing subsequent tests.
-     */
-    private static class CatchFeatureFailuresRunListener extends AbstractRunListener {
-
-        private final FeatureInfo feature
-        private final String[] iterationMatchers
-        private final RecordFailuresInterceptor failuresInterceptor
-        private final FeatureFilterInterceptor fixturesInterceptor
 
-        CatchFeatureFailuresRunListener(FeatureInfo feature, String[] iterationMatchers) {
-            this.feature = feature
-            this.iterationMatchers = iterationMatchers
-            this.failuresInterceptor = new RecordFailuresInterceptor(iterationMatchers)
-            this.fixturesInterceptor = new FeatureFilterInterceptor(feature, iterationMatchers, failuresInterceptor)
+        if (GradleContextualExecuter.isNotInstant()) {
+            return
         }
 
-        @Override
-        void beforeSpec(SpecInfo spec) {
-            spec.specsTopToBottom*.each { s ->
-                s.setupMethods*.interceptors*.add(0, fixturesInterceptor)
-                s.cleanupMethods*.interceptors*.add(0, fixturesInterceptor)
-            }
+        if (!isEnabledSpec(annotation, feature)) {
+            return
         }
 
-        @Override
-        void beforeFeature(FeatureInfo featureInfo) {
-            if (featureInfo == feature) {
-                featureInfo.featureMethod.interceptors.add(0, failuresInterceptor)
-            }
+        if (annotation.skip() == FLAKY) {
+            feature.skipped = true
+            return
         }
 
-        @Override
-        void beforeIteration(IterationInfo iteration) {
-            if (iteration.feature == feature) {
-                iteration.feature.iterationInterceptors.add(0, failuresInterceptor)
-            }
+        if (feature.isParameterized()) {
```

Comment:
These changes drop one feature: being able to skip some iterations according to `iterationsMatchers`.
Fortunately we don't have tests annotated as such anymore. Let's revisit when/if we need it.

Reply:
Yes. I tried to keep only what was strictly necessary to make the current test suite pass. It's complicated enough as it is.

Reply:
I might have misunderstood your comment. `iterationMatchers` and `bottomSpecs` are still honoured.

Reply:
I'm not sure. Is it for skipping?

Reply:
Ah, skipping. Right, no, the combination of `FAILS_TO_CLEANUP` with `iterationMatchers` is not honoured.

Reply:
In any case, you did good by only keeping what was strictly necessary 👍 
That thing got monstrous at the beginning with so many cases to handle.